test/v1/test_bucket.py

```diff
@@ -357,17 +357,17 @@ def test_ignores_subdirectory(self):
         ]
         self.assertEqual(expected, actual)
 
-    # def test_no_second_response(self):
-    #     data = six.b('hello world')
-    #     file_id = self.bucket.upload_bytes(data, 'a/b').id_
-    #     self.bucket.upload_bytes(data, 'a/b/c')
-
-    #     expected = [(file_id, 'a/b', 11, 'upload')]
-    #     actual = [
-    #         (info.id_, info.file_name, info.size, info.action)
-    #         for info in self.bucket.list_file_versions('a/b', fetch_count=1)
-    #     ]
-    #     self.assertEqual(expected, actual)
+    def test_all_versions_in_response(self):
+        data = six.b('hello world')
+        file_id = self.bucket.upload_bytes(data, 'a/b').id_
+        self.bucket.upload_bytes(data, 'a/b/c')
```

Comment:
```suggestion
        self.bucket.upload_bytes(data, 'a/b')
```
without the suggested change, this will be skipped as in the test above, due to the file being a subdirectory, not due to the fetch_count limit being equal to 1

Reply:
If we use the same name then it skips the point of the test, there will be a second version in the next response. I could use something like `a/bb` to do the same thing without the subdirectory bit.

I'm not sure I see why the subdirectory has any effect though, can you explain more?

Reply:
Right. In order to test whether the limit is passed correctly we'd have to mock up the session, I'd say there is no point in doing that. Lets leave it.